src/main/java/com/google/starfish/servlets/FavoriteNoteServlet.java

```diff
@@ -0,0 +1,135 @@
+  private NoteService noteService = new NoteService();
+  private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
+  private UserService userService = new UserService();
+  private final String COOKIE_NAME = "SFCookie";
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException{
+    res.setContentType("application/json");
+    if(!validateUser(req)) {
+      res.getWriter().println(false);
+      return;
+    }
+    HttpSession activeSession = req.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+    Long noteId = Long.valueOf(req.getParameter("note_id"));
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+    try(Connection conn = pool.getConnection()) {
+      String stmt =
+          "SELECT EXISTS "
+        + "(SELECT * "
+        + "FROM favorite_notes "
+        + "WHERE user_id=? "
+        + "AND note_id=?);";
+
+      try (PreparedStatement favStmt = conn.prepareStatement(stmt)) {
+        favStmt.setString(1, userId);
+        favStmt.setLong(2, noteId);
+        ResultSet rs = favStmt.executeQuery();
+        rs.next();
+        res.getWriter().println(rs.getBoolean(1));
+      }
+    } catch (SQLException ex) {
+      System.err.print(ex);
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    if(!validateUser(req)) {
+      res.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    HttpSession activeSession = req.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+    Long noteId = Long.valueOf(req.getParameter("note_id"));
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
+
+    try (Connection conn = pool.getConnection()) {
+      favoriteNoteService.insertFavoriteNote(pool, noteId, userId);
+      Note downloadedNote = noteService.getNoteByNoteId(pool, noteId);
+      String authorId = downloadedNote.getAuthorId();
+      userService.increasePointsOnFavorite(pool, authorId);
+    } catch (SQLException ex) {
+      System.err.print(ex);
+    }
+  }
+
+  @Override
+  public void doDelete(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    if(!validateUser(req)) {
+      res.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    HttpSession activeSession = req.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+    Long noteId = Long.valueOf(req.getParameter("note_id"));
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+    try (Connection conn = pool.getConnection()) {
+      favoriteNoteService.deleteRowByCompoundId(pool, noteId, userId);
+      Note downloadedNote = noteService.getNoteByNoteId(pool, noteId);
+      String authorId = downloadedNote.getAuthorId();
```

Comment:
nit: don't think `downloadedNote` is the right name here. Prefer if this was just 1 line like `String authorId = noteService.getNoteByNoteId(pool, noteId).getAuthorId();`.

Reply:
My bad I think I might've copy pasted code from the download-logic. I'll change that!